app/commands.js

```diff
@@ -14,34 +14,143 @@ import {
+        const sourceRepo = payload.repository.name;
+        const reason =
+          closeMatches.length > 1 && closeMatches[1] === "not-planned"
+            ? "NOT_PLANNED"
+            : "COMPLETED";
+        console.log(
+          `${id} Closing issue ${
+            payload.issue.html_url
+          }, reason: ${reason} ${actorRequest(payload)}`
+        );
+        await closeIssue(authToken, sourceRepo, payload.issue.node_id, reason);
+      },
     },
     reopen: {
       matches: reopenMatcher(commentBody),
       enabled: (config, octokit) => reopenEnabled(config, octokit),
+      run: async (id, payload, authToken) => {
+        const sourceRepo = payload.repository.name;
+        console.log(
+          `${id} Re-opening issue ${payload.issue.html_url} ${actorRequest(
+            payload
+          )}`
+        );
+        await reopenIssue(authToken, sourceRepo, payload.issue.node_id);
+      },
     },
     label: {
       matches: labelMatcher(commentBody),
-      enabled: (octokit, config, labels) =>
-        labelEnabled(octokit, config, labels),
+      enabled: (octokit, config, matches) => {
+        const labels = extractCommaSeparated(matches[1]);
+        return labelEnabled(octokit, config, labels);
+      },
+      run: async (id, payload, authToken, matches) => {
+        const labels = extractCommaSeparated(matches[1]);
+        const sourceRepo = payload.repository.name;
+
+        console.log(
+          `${id} Labeling issue ${
+            payload.issue.html_url
+          } with labels ${labels} ${actorRequest(payload)}`
+        );
+        await addLabel(
+          authToken,
+          payload.repository.owner.login,
+          sourceRepo,
+          payload.issue.node_id,
+          labels
+        );
+      },
     },
-    "remove-label": {
+    removeLabel: {
       matches: removeLabelMatcher(commentBody),
-      enabled: (config, octokit, labels) =>
-        removeLabelEnabled(config, octokit, labels),
+      enabled: (config, octokit, matches) => {
+        const labels = extractCommaSeparated(matches[1]);
+        return removeLabelEnabled(config, octokit, labels);
+      },
+      run: async (id, payload, authToken, matches) => {
+        const sourceRepo = payload.repository.name;
+        const labels = extractCommaSeparated(matches[1]);
+        console.log(
+          `${id} Removing label(s) from issue ${
+            payload.issue.html_url
+          }, labels ${labels} ${actorRequest(payload)}`
+        );
+        await removeLabel(
+          authToken,
+          payload.repository.owner.login,
+          sourceRepo,
+          payload.issue.node_id,
+          labels
+        );
+      },
     },
     reviewer: {
       matches: reviewerMatcher(commentBody),
       enabled: (config, octokit) => reviewerEnabled(config, octokit),
+      run: async (id, payload, authToken, matches) => {
+        const reviewerMatches = matches[1];
```

Comment:
```suggestion
        const reviewerMatches = this.matches;
```

This should work and avoids having to pass matches.

Reply:
ah nice thanks

Reply:
FYI That should also work for enabled.

so `run` should be able to use `this.enabled(...)`

Reply:
probably not needed though as then every run would need to handle the reporting when disabled.

Reply:
Maybe worth writing a class: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes

This should make it more obvious that you have access to run and enabled 😉

Reply:
I wasn't able to make accessing `matches` from there work, but I've converted it to a class now and it's working

Reply:
Been a long time I worked with object literals but ya a class makes it easier to extend new commands.